src/api/folders.js

```diff
@@ -0,0 +1,52 @@
+/*
+ Copyright (C) 2021 Aman Dwivedi (aman.dwivedi5@gmail.com)
+ 
+ SPDX-License-Identifier: GPL-2.0
+
+ This program is free software; you can redistribute it and/or
+ modify it under the terms of the GNU General Public License
+ version 2 as published by the Free Software Foundation.
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License along
+ with this program; if not, write to the Free Software Foundation, Inc.,
+ 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+import { endpoints } from "../constants/endpoints";
+import sendRequest from "./sendRequest";
+import { getToken } from "../shared/authHelper";
+
+export const getAllFoldersApi = () => {
+  const url = endpoints.folders.getAll();
+  return sendRequest({
+    url,
+    method: "GET",
+    credentials: "include",
+    headers: {
+      Authorization: getToken(),
```

Comment:
Do we need to use `async` here? It caused issue for me with authorization as undefined

https://github.com/fossology/FOSSologyUI/pull/34#issuecomment-869494031

Reply:
I am not sure about it. I believe there should not be `async`. Also for me it is working without async.
Though I believe that there is a need for an additional check that if the token is revoked during the user session.
Then we do need to call for token again.
Could you please confirm that it was not working for you even after logout and then again logging in (without async)?

Reply:
Hi @GMishx 
I have added the `async` keyword. Do let me know if this is working for you now.
Although my code editor still says that `async` and `await` has no effect in this case.
![image](https://user-images.githubusercontent.com/54680709/123734224-8ca3e500-d8ba-11eb-9220-ae5be84af94b.png)

Reply:
Yeah for me also it is working without async, but later I have taken a another variable like,
`const token = await getToken();` and assinged token to Authorization in the headers.

Reply:
I am not sure about it, not very good with JS. But I faced this issue in #34 and now it seems to be working good.